lua/pixelui/core/cl_images.lua

```diff
@@ -21,26 +21,30 @@ local queue = {}
 local useProxy = false
 
 file.CreateDir("pixel")
+file.CreateDir(PIXEL.DownloadPath)
 
 local function processQueue()
     if queue[1] then
-        local id, matSettings, callback = unpack(queue[1])
+        local url, matSettings, callback = unpack(queue[1])
 
-        http.Fetch((useProxy and "https://proxy.duckduckgo.com/iu/?u=https://i.imgur.com/" or "https://i.imgur.com/") .. id .. ".png",
+        local filePath = PIXEL.DownloadPath .. "/" .. url
+        url = "https://" .. url
```

Comment:
We can't assume everything supports HTTPS, many servers deliberately disable it for their asset subdomains in order to work with Awesomium.